extensions/ql-vscode/src/extension.ts

```diff
@@ -1529,6 +1530,16 @@ async function activateWithInstalledDistribution(
   };
 }
 
+function addUnhandledRejectionListener() {
+  process.addListener("unhandledRejection", (error: unknown) => {
+    void showAndLogExceptionWithTelemetry(
+      redactableError(asError(error))`Unhandled error: ${getErrorMessage(
+        error,
+      )}`,
+    );
```

Comment:
What happens if `showAndLogExceptionWithTelemetry` throws? It will be an `unhandledRejection`, I think. And this would cause a loop of this handler being called. Unlikely to happen, but if it happens, it will be bad.

I think it would be best to add a `.then(...)` handler here, with a catch argument that just logs to. Or, you can just be _really_ sure that `showAndLogExceptionWithTelemetry` will never throw.

Reply:
Good point. I think we can add a `catch` to `showAndLogExceptionWithTelemetry` which then logs in a much more simple way. Or alternatively we could just fully ignore it in this case. That wouldn't put us in a worse place than we were before this PR.

Reply:
Actually it seems NodeJS thought of this:
> Exceptions thrown from within the event handler will not be caught. Instead the process will exit with a non-zero exit code and the stack trace will be printed. This is to avoid infinite recursion.

From https://nodejs.org/api/process.html#event-uncaughtexception